File: PHP/6/example6.php
#!/usr/bin/env php

<?php
$outfile = "out.txt";

$text = "aldsfkjlsadfkj\nalsdfkjalsdfjk\nalsdkjflasdfjk\nlasdkjlsadkjf\n";

echo "Writing to outfile ... \n";

$fh = fopen($outfile, "w") or die("Cannot open file: $outfile.\n");
fwrite($fh, $text);
fclose($fh);

echo "Reading the file line by line ...\n";
$fh = fopen($outfile, "r") or die("Cannot open file: $outfile.\n");
while (!feof($fh))
{
    $line = fgets($fh);
    echo "$line";
}
fclose($fh);

echo "Adding more text to outfile ...\n";
$more_text = "alsdkjfalfdkj\nalsdfjldfkj\n";

$fh = fopen($outfile, "a") or die("Cannot open file: $outfile.\n");
fwrite($fh, $more_text);
fclose($fh);

echo "Printing out all text from outfile ...\n";
$out = file_get_contents($outfile);
echo $out;

File: Perl/6/example6.pl
#!/usr/bin/env perl

use strict;
use warnings;
use File::Slurp;

my $outfile = "out.txt";

my $text = "aldsfkjlsadfkj\nalsdfkjalsdfjk\nalsdkjflasdfjk\nlasdkjlsadkjf\n";

print "Writing to outfile ...\n";
open(my $fh, '>', $outfile) or die ("Could not open file: $outfile");
print $fh $text;
close $fh;

print "Reading from outfile ...\n";
open($fh, '<:encoding(UTF-8)', $outfile) or die ("Could not open file: $outfile");
while (my $row = <$fh>)
{
    chomp $row;
    print "$row\n";
}
close $fh;

print "Adding more text to outfile ...\n";
my $more_text = "alsdkjfalfdkj\nalsdfjldfkj\n";
open($fh, '>>', $outfile) or die ("Could not open file: $outfile");
print $fh $more_text;
close $fh;

print "Opening outfile to read the entire text ...\n";
my $out = read_file($outfile);
print $out;
